packages/fxa-payments-server/src/lib/errors.ts

```diff
@@ -111,14 +128,24 @@ const paymentErrors2 = [
 
 const paymentErrors3 = ['general-paypal-error'];
 
-cardErrors.forEach((k) => (errorMessageIndex[k] = CARD_ERROR));
-basicErrors.forEach((k) => (errorMessageIndex[k] = BASIC_ERROR));
-paymentErrors1.forEach((k) => (errorMessageIndex[k] = PAYMENT_ERROR_1));
-paymentErrors2.forEach((k) => (errorMessageIndex[k] = PAYMENT_ERROR_2));
-paymentErrors3.forEach((k) => (errorMessageIndex[k] = PAYMENT_ERROR_3));
+cardErrors.forEach((k) => (errorToErrorMessageMap[k] = CARD_ERROR));
+basicErrors.forEach((k) => (errorToErrorMessageMap[k] = BASIC_ERROR));
+paymentErrors1.forEach((k) => (errorToErrorMessageMap[k] = PAYMENT_ERROR_1));
+paymentErrors2.forEach((k) => (errorToErrorMessageMap[k] = PAYMENT_ERROR_2));
+paymentErrors3.forEach((k) => (errorToErrorMessageMap[k] = PAYMENT_ERROR_3));
 
-function getErrorMessage(type: string) {
-  return errorMessageIndex[type] ? errorMessageIndex[type] : BASIC_ERROR;
+function getErrorMessage(error: undefined | StripeError | GeneralError) {
+  if (!error) {
+    return BASIC_ERROR;
+  }
+  let lookup = 'UNKNOWN';
+  // Handle case where we need to lookup by message (e.g. currency error)
+  if (error.message && errorToErrorMessageMap[error.message]) {
+    lookup = error.message;
+  } else if (error.code && errorToErrorMessageMap[error.code]) {
+    lookup = error.code;
+  }
```

Comment:
I guess we have to add this `message` property, since this check is on the server, and the `code` is an HTTP response code rather than a Stripe error code?

Reply:
I'm not sure what you mean bianca, can you elaborate? Or we can talk about it live.